### ISSUE TITLE
toContainHTML failure output is not useful

### ISSUE THREAD

**Issue by ratbeard** [testing-library/jest-dom#98]
* `jest-dom` version: 3.1.3
* `node` version: v10.11.0
* `npm` (or `yarn`) version: 6.9.0

### What you did:

```js
const container = render(<SubmitButton>)
expect(container).toContainHtml('color: blue')
```

### What happened:

<img width="420" alt=" 2019-04-25 at 3 27 53 PM" src="https://user-images.githubusercontent.com/28356/56766644-83b1a900-676f-11e9-8797-4f396af6e37e.png">

### Problem description:

The output failure message does not contain any useful information for tracking down the reason why it failed.  

I have to convert it to something like `expect(container.innerHTML).toContain('color: blue')` to be able to debug it, making the custom matcher less efficient than just using the simpler builtin option.

Note, I saw theres a style matcher that could be used in this specific test, however the problem applies to any call to this matcher.

### Suggested solution:

Pretty sure it just needs to change to `cloneNode(true)` at https://github.com/testing-library/jest-dom/blob/master/src/to-contain-html.js#L14 for it to print out all the nodes.  

That fix would make it debuggable.  To make it better than the output of `toContain()` it could be pretty-printed as indented and syntax highlighted html 😃 - any chance that has been considered or used by other matchers?  

Here is the toContain() output for reference.  You can imagine its much harder to debug a larger dom node

<img width="597" alt=" 2019-04-25 at 3 40 59 PM" src="https://user-images.githubusercontent.com/28356/56767258-e0619380-6770-11e9-8601-04883fe82460.png">

**Comment by gnapse** [comment]
Thanks for reporting this.

If you want to assert that an element has some style, you should prefer using [`.toHaveStyle`](https://github.com/testing-library/jest-dom#tohavestyle). The matcher you're trying to use is not meant for that. I know you mentioned it, but I wanted to stress that point, because fixing this should not be something encourage checking styles in that way.

In addition to not being intended to test for inline css styles, `toContainHTML` is not even really recommended in the general case, if you know the structure of the html that you want to test, you better test it in a more semantical way. See this disclaimer also found in the README about `.toContainHTML`:

> It could be useful in situations where the code being tested renders html that was obtained from an external source, and you want to validate that that html code was used as intended.

However, with all that being said, I do agree with you that if switching that line of code would give better errors, then we should do it. Can you contribute it? Otherwise, I may get to it at some point in the next few days, if it's really that simple.

**Comment by ratbeard** [comment]
Thanks for the reply.  

I'm kind of on the fence if it's better to take the loosey-goosey html approach vs using more precise finders like `toHave{Style,Class,Attribute}`, specifically when the attributes effects a sub-element of the component under test.  The loosey-goosey approach gives you slightly less accuracy on the one hand, but is less brittle as markup changes.  Most of these tests impact visual rendering, so the test could stay green that a div got a passed-in class applied even though a css refactor means the element has a height of 0 on screen now - it's really visual component snapshots you need to be using to guarantee those things.  I've got a TODO to implement those!

Anyways thats just my current thought process, kinda curious if other people have thought that or not.

> However, with all that being said, I do agree with you that if switching that line of code would give better errors, then we should do it. Can you contribute it? Otherwise, I may get to it at some point in the next few days, if it's really that simple.

Yep, I can do that 👍

**Comment by afontcu** [comment]
I believe this can be closed thanks to #100

**Comment by gnapse** [comment]
Thanks for pointing that out
